crates/runtime/src/dataaccelerator/sqlite.rs

```diff
@@ -75,6 +76,12 @@ pub struct SqliteAccelerator {
 impl SqliteAccelerator {
     #[must_use]
     pub fn new() -> Self {
+        // Initialize the decimal extension for SQLite
+        //
+        // SAFETY: This is safe because sqlite3_decimal_init is a valid function pointer.
+        unsafe {
+            sqlite3_auto_extension(Some(sqlite3_decimal_init));
```

Comment:
shouldn't this belong in `datafusion-table-providers`, where the connections are managed?

Reply:
We can't assume that everyone using the `datafusion-table-providers` will be using a version of rusqlite with the decimal support compiled in.